internal/dag/dag.go

```diff
@@ -491,10 +492,11 @@ type Route struct {
 
 func (r *Route) Prefix() string { return r.path }
 
-func (r *Route) addService(s *Service) {
+func (r *Route) addService(s *Service, weight *int) {
 	if r.services == nil {
 		r.services = make(map[portmeta]*Service)
 	}
+	s.Weight = weight
```

Comment:
I don't think this field belongs on the service object. Unlike https://github.com/heptio/contour/pull/498#discussion_r199679683 I think that a service referenced with two different weights is a different service. This mirrors the way envoy models service weighting; the weighted configuration goes into the RDS configuration--you'll never find anything to do with a cluster's weight on the cluster object in CDS.

I think we can live with this for beta.1, but please raise an issue to address it for 0.6 final.

Reply:
Yup that's a good point, let me look to refactor it a bit.

Reply:
> I think that a service referenced with two different weights is a different service
Previous conversations around this lead to this being an error. Are we going to support different weights / lb algo / healthchecks or are those all errors?